src/cpp/rtps/builtin/discovery/participant/PDPServerListener.cpp

```diff
@@ -304,6 +313,14 @@ void PDPServerListener::onNewCacheChangeAdded(
                     pdp_server()->assignRemoteEndpoints(pdata);
                 }
             }
+            // Case ParticipantProxyData already exists but was known remotly and now must be local
+            else if (is_local && !was_local)
+            {
+                // Realease PDP mutex
+                lock.unlock();
+
+                pdp_server()->assignRemoteEndpoints(pdata);
```

Comment:
Is the database *locality* of the participant updated too? It must be so to avoid try to rematch the endpoints on later announcements.

Reply:
Nope, the method `DiscoveryDataBase::create_participant_from_change_` must be modified to account for *locality* change even if the *sequence number* of the `DATA(p)` is the same kept in the database.

Reply:
:+1: